tests/sqlparser_common.rs

```diff
@@ -1647,6 +1647,46 @@ fn parse_cast() {
         },
         expr_from_projection(only(&select.projection))
     );
+
+    let sql = "SELECT CAST(id AS CLOB(50)) FROM customer";
+    let select = verified_only_select(sql);
+    assert_eq!(
+        &Expr::Cast {
+            expr: Box::new(Expr::Identifier(Ident::new("id"))),
+            data_type: DataType::Clob(50)
+        },
+        expr_from_projection(only(&select.projection))
+    );
```

Comment:
@alamb I added test cases for `parse_cast` with these data types since I could not find test fn `parse_datatype`.. Thanks for review.